elide-core/src/main/java/com/yahoo/elide/core/DataStoreTransaction.java

```diff
@@ -189,7 +189,8 @@ default void preCommit() {
             EntityDictionary dictionary,
             Optional<FilterExpression> filterExpression,
             Sorting sorting,
-            Pagination pagination
+            Pagination pagination,
+            String relationNameForSortAndPage
```

Comment:
Instead of pushing this responsibility to the data store to differentiate when to paginate or when to not, there should be versions of getRelation - one with pagination and one without (or Pagination could be Optional).  The persistent resource should know when to call which interface.